src/module.c

```diff
@@ -3823,6 +3823,29 @@ RedisModuleKey *RM_OpenKey(RedisModuleCtx *ctx, robj *keyname, int mode) {
     return kp;
 }
 
+/**
+ * Returns the full OpenKey modes mask, using the return value
+ * the module can check if a certain set of OpenKey modes are supported
+ * by the redis server version in use.
+ * Example:
+ *
+ *        int supportedMode = RM_GetOpenKeyModesAll();
+ *        if (supportedMode & REDISMODULE_OPEN_KEY_NOTOUCH) {
+ *              // REDISMODULE_OPEN_KEY_NOTOUCH is supported
+ *        } else{
+ *              // REDISMODULE_OPEN_KEY_NOTOUCH is not supported
+ *        }
+ */
+int RM_GetOpenKeyModesAll() {
+    return REDISMODULE_READ               |
+           REDISMODULE_WRITE              |
+           REDISMODULE_OPEN_KEY_NOTOUCH   |
+           REDISMODULE_OPEN_KEY_NONOTIFY  |
+           REDISMODULE_OPEN_KEY_NOSTATS   |
+           REDISMODULE_OPEN_KEY_NOEXPIRE  |
+           REDISMODULE_OPEN_KEY_NOEFFECTS;
```

Comment:
i fear that leaving this here in the C file risks forgetting to update it when a new option is added.
we can maybe add a comment in the header about this function, or better yet, move that list as a single define to be placed next to the flags in the header.
like we do with `_REDISMODULE_OPTIONS_FLAGS_NEXT - 1`

Reply:
The problem with the next approach here is that `REDISMODULE_READ` and `REDISMODULE_WRITE` have values `(1<<0)` and `(1<<1)` respectively, but then `REDISMODULE_OPEN_KEY_NOTOUCH` jump to `(1<<16)` and this was already been this way so we can change it.

Reply:
right, maybe we can do some trick of first and last in each group and combine them, but even if we can't, defining that list of all flags in the header can be better than defining it in the C file, since it'll closer to the place where new ones are added.
and if we don't do that, then at least add a comment with reminder not to forget this function.

Reply:
Updated, let me know what you think.